scripts/coccinelle/try_parse.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+
+# Echo the name of every argument of this script that is not "perfect"
+# according to coccinelle's --parse-c.
+
+top="$(dirname "$0")/../.."
+
+for fn in "$@"; do
+
+    if spatch -macro_file_builtins "$top"/scripts/coccinelle/tor-coccinelle.h \
+              -I "$top" -I "$top"/src -I "$top"/ext --parse-c "$fn" \
+              2>/dev/null | grep "perfect = 1" > /dev/null; then
+        : # it's perfect
+    else
+        echo "$fn"
```

Comment:
Can you make this script exit with a non-zero exit status if it finds any imperfect files?
That will let us call it from make and the git hooks in trac 31919.

Reply:
Can you also make try_parse.sh show the parsing errors, so they can be fixed?
If this verbose mode needs to be activated by an env var or argument, that's ok.